src/core/NewDomain.ml

```diff
@@ -482,7 +483,7 @@ struct
 
     | Neu info ->
       let neu = {head = NCoe {r; r'; ty = Abs (x, info.neu); cap}; frames = Emp} in
-      let ty = Val.hsubst r' x rel tyx in
+      let ty = val_hsubst r' x (Rel.hide' x rel) tyx in
```

Comment:
@jonsterling I believe there was a bug when `Rel.hide` was missing. (Unless you claimed `x` is always fresh.)

Reply:
SHould that just be built into Val.hsubst?

Reply:
No, because in general `subst r x (hide x rel)` is different from `subst r x rel`.
```
subst 1 x (hide x {x=y}) = {}
subst 1 x {x=y} = {y=1}
```

Reply:
Oh sorry, I misread the code. Now I see, and I think your change is correct.